src/artm/core/data_loader.cc

```diff
@@ -342,6 +364,11 @@ void RemoteDataLoader::ThreadFunction() {
         break;
       }
 
+      if (instance()->schema()->config().online_batch_processing()) {
```

Comment:
Please open a new issue to remove DataLoaderThread in online_processing_mode. However I think it might be really simple. The only thing you need to do is comment out this lines:

  // Keep this at the last action in constructor.
  // http://stackoverflow.com/questions/15751618/initialize-boost-thread-in-object-constructor
  boost::thread t(&LocalDataLoader::ThreadFunction, this);
  thread_.swap(t);

Then thread_ will refer to what boost calls 'not-a-thread'. Check http://www.boost.org/doc/libs/1_55_0/doc/html/thread/thread_management.html for the documentation.